### DISCUSSION TITLE
Streaming data into perspective-python Table

### DISCUSSION THREAD

**Discussion by robambalu** [perspective-dev/perspective#1363] [Q&A] [answered]
## Support Question

Say I had some event based processing system that does "time series processing" and generates ticks of data on some engine thread in realtime.  I want the results of this "computation graph" to feed into a python perspective table.
My main question is, say I want to join these technologies in a single process.  It seems like I run python perspective on its own tornado ioloop thread and push data to it from my graph computation thread.  Assuming that is correct, the question is if I can do table update_data calls from the non-perspective thread or should I be pushing the events to the ioloop thread by scheduling events into it.

*subliminal messaging not an accident

**Comment by robambalu** [comment]
Any thoughts?

**Comment by timkpaine** [comment]
are we talking real threads or python threads?

**Comment by robambalu** [comment]
I will have a thread running my engine generating events that i need to feed into perspective-python.  Not sure what the diff is if its "real" thread or python thread, but lets say its a python thread for this purposes ( so under GIL )

**Comment by timkpaine** [comment, marked as answer]
You should be able to do it anywhere so long as you give perspective time to do stuff, your update call needs the GIL but you don't need tornado or an ioloop explicitly. For example, perspective plugs into tornado [here](https://github.com/finos/perspective/blob/f9217fa611456e53238a1efa9d75a879d55d2f70/python/perspective/perspective/tornado_handler/tornado_handler.py#L16) but it could've been any other engine in the same manner. @sc1f can comment more

**Comment by sc1f** [comment]
Calling `update()` should work from anywhere in your event handler.
